File: articles/futures-literacy-and-anticipatory-capacity/fortran/capacity_score.f90
program capacity_score
  implicit none
  real :: values(7)
  real :: weights(7)
  real :: score

  values = (/0.82, 0.84, 0.88, 0.90, 0.72, 0.84, 0.80/)
  weights = (/0.15, 0.15, 0.17, 0.13, 0.17, 0.13, 0.10/)
  score = sum(values * weights)

  print *, "Futures-Literate Organization anticipatory capacity:", score
end program capacity_score

File: articles/futures-literacy-and-anticipatory-capacity/cpp/capacity_score.cpp
#include <iostream>
#include <vector>

double weighted_score(const std::vector<double>& values, const std::vector<double>& weights) {
    double score = 0.0;
    for (std::size_t i = 0; i < values.size(); ++i) {
        score += values[i] * weights[i];
    }
    return score;
}

int main() {
    std::vector<double> values = {0.82, 0.84, 0.88, 0.90, 0.72, 0.84, 0.80};
    std::vector<double> weights = {0.15, 0.15, 0.17, 0.13, 0.17, 0.13, 0.10};
    std::cout << "Futures-Literate Organization anticipatory capacity="
              << weighted_score(values, weights) << "\n";
    return 0;
}
